content/docs/guides/basic-auth.md

```diff
@@ -0,0 +1,84 @@
+
+NOTE: Although basic auth connections *to* Prometheus instances are not supported, TLS is supported for connections *from* Prometheus instances to [scrape targets](../prometheus/latest/configuration/configuration/#<scrape_config>).
+
+## nginx example
+
+Let's say that you want to run a Prometheus instance behind an [nginx](https://www.nginx.com/) server running on `localhost:12321`, and for all Prometheus endpoints to be available via the `/prometheus` endpoint. The full URL for Prometheus' `/metrics` endpoint would thus be:
+
+```
+http://localhost:12321/prometheus/metrics
+```
+
+Let's also say that you want to require a username and password from all users accessing the Prometheus instance. For this example, use `admin` as the username and choose any password you'd like.
+
+First, create a `.htpasswd` file to store the username/password using the [`htpasswd`](https://httpd.apache.org/docs/2.4/programs/htpasswd.html) tool and store it in the `/etc/nginx` directory:
+
+```bash
+mkdir -p /etc/nginx
+htpasswd -c /etc/nginx/.htpasswd admin
+```
+
+NOTE: This example uses `/etc/nginx` as the location of the nginx configuration files, including the `.htpasswd` file, but this will vary based on the installation. Other [common nginx config directories](http://nginx.org/en/docs/beginners_guide.html) include `/usr/local/nginx/conf` and `/usr/local/etc/nginx`.
+
+## nginx configuration
+
+Below is an example [`nginx.conf`](https://www.nginx.com/resources/wiki/start/topics/examples/full/) configuration file (stored at `/etc/nginx/.htpasswd`). With this configuration, nginx will enforce basic auth for all connections to the `/prometheus` endpoint (which proxies to Prometheus):
+
+```conf
+http {
+    server {
+        listen 12321;
+
+        location /prometheus {
+            auth_basic           "Prometheus";
+            auth_basic_user_file /etc/nginx/.htpasswd;
+
+            proxy_pass           http://localhost:9090/;
+        }
+    }
+}
+
+events {}
+```
+
+Start nginx as root (since nginx will need to bind to port 443):
+
+```bash
+sudo nginx -c /etc/nginx/nginx.conf
+```
+
+## Prometheus configuration
+
+When running Prometheus behind the nginx proxy, you'll need to set the external URL to `http://localhost:12321/prometheus` and the route prefix to `/`:
+
+```bash
+prometheus \
+  --config.file=/path/to/prometheus.yml \
+  --web.external-url=http://localhost:12321/prometheus \
+  --web.route-prefix="/"
+```
+
+## Testing
+
+You can use cURL to interact with your local nginx/Prometheus setup. The following request will return a `401 Unauthorized` response, because you've failed to supply a valid username and password, and also bear a `WWW-Authenticate: Basic realm="Prometheus"` header:
```

Comment:
It could be clearer that the WWW-Authenticate header is coming from the server, it currently reads as though it's meant to come from the client. This would only be shown if -v was passed to curl.